taichi/ir/control_flow_graph.cpp

```diff
@@ -275,7 +275,20 @@ bool CFGNode::store_to_load_forwarding(bool after_lower_access) {
       }
     } else if (auto global_load = stmt->cast<GlobalLoadStmt>()) {
       if (!after_lower_access) {
-        result = get_store_forwarding_data(global_load->src, i);
+        if (auto global_ptr = global_load->src->cast<GlobalPtrStmt>()) {
+          TI_ASSERT(global_ptr->width() == 1);
+          auto snodes = global_ptr->snodes;
+          if (!snodes[0]->has_adjoint()) {
+            // No adjoint SNode. Do the store forwarding
+            // otherwise skipping the store forwarding to keep the global
+            // load chain,
+            // so that the grad of intermidiate variable can be computed
+            // by GlobalLoadStmt
+            result = get_store_forwarding_data(global_load->src, i);
```

Comment:
can we simplify the logic here so that we write `get_store_forwarding_data(global_load->src, i)` only once?